module/addRemove.js

```diff
@@ -0,0 +1,103 @@
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let editing = -1;
+const toDoInput = document.querySelector('#new-list');
+const todoListElement = document.querySelector('.to-do-list');
+
+const saveToDos = () => {
+  const todovalue = toDoInput.value;
+  const isDuplicate = todos.some((todo) => todo.value === todovalue);
+  if (todovalue === '') {
+    return;
+  } else if (isDuplicate) {
+    return;
+  } else {
+    if (editing >= 0) {
+      todos = todos.map((todo, index) => {
+        return {
+          value: index === editing ? todovalue : todo.value,
+          checked: todo.checked,
+          color: todo.color,
+          index: index,
+        };
+      });
+      editing = -1;
+    } else {
+      todos.push({
+        value: todovalue,
+        checked: false,
+        color: '#' + Math.floor(Math.random() * 16777215).toString(16),
+        index: todovalue.id,
+      });
+      toDoInput.value = '';
+    }
+  }
+};
+const displayTodos = () => {
+  if (todos.length === 0) {
+    todoListElement.innerHTML =
+      '<center>Nothing to do yet! please add to do list </center>';
+    return;
+  }
+  todoListElement.innerHTML = '';
+  todos.forEach((todo, index) => {
+    todoListElement.innerHTML += `
+      <div class="todo" id="${index}">
+      <input type="checkbox" class="chech-box" data-action="check" />
+      <p class="list-element" data-action="check" >${todo.value}</p>
+      <i class="fa-regular fa-pen-to-square" data-action="edit" ></i>
+      <i class="fa-sharp fa-solid fa-trash" data-action="delete"></i>
+    </div>
+        `;
+  });
+};
+
+todoListElement.addEventListener('click', (event) => {
+  const target = event.target;
+  const parentElement = target.parentNode;
+
+  if (parentElement.className !== 'to-do-list') {
+    const todo = parentElement;
+    const todoId = Number(todo.id);
+    const action = target.dataset.action;
+    action === 'check' && checkTodo(todoId);
+    action === 'edit' && editTodo(todoId);
+    action === 'delete' && deleteTodo(todoId);
+  }
+});
+
+const checkTodo = (todoId) => {
+  todos = todos.map((todo, index) => {
+    if (index === todoId) {
+      return {
+        value: todo.value,
+        color: todo.color,
+        checked: !todo.checked,
+        index: index,
+      };
+    } else {
+      return {
+        value: todo.value,
+        color: todo.color,
+        checked: todo.checked,
+        index: index,
+      };
+    }
+  });
+  displayTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+const editTodo = (todoId) => {
+  toDoInput.value = todos[todoId].value;
+  editing = todoId;
+};
+
+const deleteTodo = (todoId) => {
+  todos.filter((todo, index) => index !== todoId);
+  editing = -1;
+  todos.length -= 1;
+  displayTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
```

Comment:
- [ ] Kindly make sure that when the user removes a task, the task that the user clicked on gets removed and not any other task. You can check this in the below GIF (might take some time to load)

![chrome-capture-2023-0-25](https://user-images.githubusercontent.com/62290545/214584063-8a6d4c5d-0087-4739-8b9f-a05303fb3f08.gif)

- [ ] Please make sure that removing a task updates all remaining items' indexes, as per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should%20update%20all%20remaining%20items%27%20indexes%2C%20so%20they%20represent%20the%20current%20list%20order%20and%20are%20unique(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..) requirement.

Reply:
- Well resolved @haftamudesta 👍🏽